### PR TITLE
Sg/export dependencies

### PR DESCRIPTION
**Pull request by elizabeth-legros**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
---
When exporting a shadergraph or subgraph with dependencies, the depended assets were not grabbed correctly. 

Fix for https://fogbugz.unity3d.com/f/cases/1210945#BugEvent.1072132109

### Checklist for PR maker
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.

### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- Other: 
Reimported all files in the test project and tried exporting a few shadergraphs

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: com.unity.shadergraph/Editor/Importers/ShaderSubGraphImporter.cs [hunk 1]
```diff
@@ -73,6 +73,16 @@ public override void OnImportAsset(AssetImportContext ctx)
             Texture2D texture = Resources.Load<Texture2D>("Icons/sg_subgraph_icon@64");
             ctx.AddObjectToAsset("MainAsset", graphAsset, texture);
             ctx.SetMainObject(graphAsset);
+
+            var metadata = ScriptableObject.CreateInstance<ShaderGraphMetadata>();
```
Comment (elizabeth-legros) [inline com.unity.shadergraph/Editor/Importers/ShaderSubGraphImporter.cs:77]: I think I meant this to be ShaderSubGraphMetadata, need to fix

Comment (alindmanUnity) [reply com.unity.shadergraph/Editor/Importers/ShaderSubGraphImporter.cs]: fixed

**Comment by alindmanUnity** [comment]
Manual testing:
- Made a shader graph that referenced a subgraph. Exported only the shader and checked subgraphs were included. 
- Nested a subgraph in a sub graph. Exported only the top level subgraph and checked that all child graphs were included. 
Added a nested sub graph to a shader graph. Export only the shader graph and checked that all nested sub graphs were included.

**Review by alindmanUnity — APPROVED** [review]

**Comment by alindmanUnity** [comment]
CI failures seem unrelated to this code change, all SG tests are passing

**Pull request merged**